packages/core/src/demo/echo.js

```diff
@@ -108,6 +150,7 @@ const messageEchoCache = {};
     const cacheOriginal = assetOriginalCache[messageId];
     if (!cacheOriginal) {
       logger.warn(`Uploaded data for message ID "${messageId} was received before the metadata."`);
```

Comment:
I think this is a case we handle in the webapp. If the preview comes before the actual text message, we save it as is. (I believe there was some stories like this in the past with mobile client + bad networking or something).
// ping @gregor